src/main/kotlin/no/skatteetaten/aurora/boober/feature/BigIpFeature.kt

```diff
@@ -20,82 +22,164 @@ class BigIpFeature(
+            }.toSet()
+
+        val legacyConfig = setOf(
             AuroraConfigFieldHandler("bigip/service"),
             AuroraConfigFieldHandler("bigip/asmPolicy"),
             AuroraConfigFieldHandler("bigip/externalHost"),
             AuroraConfigFieldHandler("bigip/oauthScopes"),
             AuroraConfigFieldHandler("bigip/apiPaths"),
             AuroraConfigFieldHandler("bigip/enabled", { it.boolean() })
         ) + findRouteAnnotationHandlers("bigip", cmd.applicationFiles, "routeAnnotations")
+
+        return multipleConfigs + legacyConfig
     }
 
     override fun validate(
         adc: AuroraDeploymentSpec,
         fullValidation: Boolean,
         context: FeatureContext
     ): List<Exception> {
-        if (adc.hasSubKeys("bigip") && adc.getOrNull<String>("bigip/service").isNullOrEmpty()) {
-            throw AuroraDeploymentSpecValidationException("bigip/service is required if any other bigip flags are set")
+        val hosts = adc.getBigIPHosts()
+        val isMultipleConfig = hosts.isNotEmpty()
+
+        val isMissingLegacyServiceConfig =
+            adc.hasSubKeys("bigip") && adc.getOrNull<String>("bigip/service").isNullOrEmpty()
+
+        val isMissingMultipleServiceConfig = hosts.any {
+            adc.getOrNull<String>("bigip/$it/service").isNullOrEmpty()
+        }
+
+        val hasConfiguredBothLegacyAndMultipleConfig = !isMissingLegacyServiceConfig && isMultipleConfig
+        if (hasConfiguredBothLegacyAndMultipleConfig) {
+            throw AuroraDeploymentSpecValidationException(Errors.BothLegacyAndMultipleConfigIsSet.message)
+        }
+
+        if (!isMultipleConfig && isMissingLegacyServiceConfig) {
+            throw AuroraDeploymentSpecValidationException(Errors.MissingLegacyService.message)
+        }
+
+        if (isMissingMultipleServiceConfig) {
+            throw AuroraDeploymentSpecValidationException(Errors.MissingMultipleService.message)
         }
+
         return emptyList()
     }
 
     override fun generate(adc: AuroraDeploymentSpec, context: FeatureContext): Set<AuroraResource> {
-        val enabled = adc.isFeatureEnabled()
-        if (!enabled) {
+        return adc.getBigIPHosts()
+            .flatMap { generateBigIPResources(it, adc) }
+            .toSet()
+    }
+
+    fun generateBigIPResources(host: String, adc: AuroraDeploymentSpec): Set<AuroraResource> {
+        if (!adc.isBigIPHostEnabled(host)) {
             return emptySet()
         }
 
-        val routeName = "${adc.name}-bigip"
+        // This is used to preserve legacy configuration
+        val isApplicationHost = host == adc.name
 
-        // dette var den gamle applicationDeploymentId som må nå være hostname
-        val routeHost = DigestUtils.sha1Hex("${adc.namespace}/${adc.name}")
+        val routeHost = if (isApplicationHost) {
+            DigestUtils.sha1Hex("${adc.namespace}/${adc.name}")
+        } else {
+            "$host-${adc.namespace}"
```

Comment:
Vi bør være veldig tydelig på at dette vil føre til nedetid for applikasjonen. Det tar vel fort opp til 10 min i konfigtjenesten.